pkg/cri/server/container_events.go

```diff
@@ -17,11 +17,16 @@
 package server
 
 import (
-	"errors"
-
 	runtime "k8s.io/cri-api/pkg/apis/runtime/v1"
 )
 
 func (c *criService) GetContainerEvents(r *runtime.GetEventsRequest, s runtime.RuntimeService_GetContainerEventsServer) error {
-	return errors.New("not implemented")
+	// TODO (https://github.com/containerd/containerd/issues/7318):
+	// replace with a real implementation that broadcasts containerEventsChan to all subscribers.
+	for event := range c.containerEventsChan {
```

Comment:
expected this to be a pub sub.. stream from the KEP

Reply:
sorry i don't quite follow. This function reads events from the buffered channel and sends it to the `RuntimeService_GetContainerEventsServer` and the client can read from it. It's basically the same as the tutorial here: https://grpc.io/docs/languages/go/basics/#server-side-streaming-rpc

Reply:
nod TODO+issue is sufficient thx